pkg/signup/signup_service.go

```diff
@@ -238,6 +242,45 @@ func (s *ServiceImpl) GetSignup(userID string) (*Signup, error) {
 	return signupResponse, nil
 }
 
+// PostVerification returns userSignup resource from the host cluster.
+// Returns the usersignup, an http code and an error if present.
+func (s *ServiceImpl) PostVerification(dailyLimit int, responseBody map[string]string, userID, code string) (*crtapi.UserSignup, int, error) {
+	// Retrieve UserSignup resource from the host cluster
+	userSignup, err := s.UserSignups.Get(userID)
+	if err != nil {
+		return nil, http.StatusNotFound, err
+	}
+
+	// get the annotation counter
+	annotationCounter := userSignup.Annotations[crtapi.UserSignupVerificationCounterAnnotationKey]
```

Comment:
This logic should be in the verification service.

Reply:
hmm not sure i understand why? the signup resource is created/modified in the signup_service. by doing this all in the verification service we are sort of bringing in some inconsistencies.

Reply:
fixed here: https://github.com/codeready-toolchain/registration-service/blob/60ca7377aabe427ed8f8e3d642380b1b8396454e/pkg/controller/signup.go#L51